src/query/api/v1/handler/prometheus/native/middleware.go

```diff
@@ -76,9 +76,7 @@ var WithInstantQueryParamsAndRangeRewriting middleware.OverrideOptions = func(
 var middlewareParseParams middleware.ParseQueryParams = func(r *http.Request, requestStart time.Time) (
 	middleware.QueryParams, error) {
 	query := r.FormValue(QueryParam)
-	if query == "" {
```

Comment:
This is no longer an issue since label APIs have no query

Reply:
not following, do they not use this?

Reply:
Nope, label APIs filter with match[] if they do any filtering:

```
match[]=<series_selector>: Repeated series selector argument that selects the series from which to read the label values. Optional.
```